api/envoy/config/core/v3/protocol.proto

```diff
@@ -313,8 +313,8 @@ message Http2ProtocolOptions {
   // of WINDOW_UPDATE frames received over the lifetime of connection exceeds the value calculated
   // using this formula::
   //
-  //     1 + 2 * (inbound_streams +
-  //              max_inbound_window_update_frames_per_data_frame_sent * outbound_data_frames)
+  //     10 * (1 + inbound_streams) +
```

Comment:
Where is `10` coming from? It's a big multiplier, without any justification stated here.

Reply:
I've changed the formula once again. Basically it adds 1 DATA frame for each new stream. This will accommodate the case of server preemptively sending WINDOW_UPDATE.
Also dropped 2 multiplier. I do not think it makes sense in the new formula.